### ISSUE TITLE
missing classes: java.util.Collections$EmptyNavigableSet

### ISSUE THREAD

**Issue by nickbreen** [spotbugs/spotbugs#4230]
There are a couple of relevant issues and PR's:
- https://github.com/spotbugs/spotbugs/issues/2140
- https://github.com/spotbugs/spotbugs/pull/2141

This appears to be another one that needs adding to that list of magic signatures.

**Describe the bug**
Some usages of `java.util.Collections.emptySortedSet()` fail with missing classes including `java.util.Collections$EmptyNavigableSet`

**To Reproduce**

```java
public class Bug {
  private final SortedSet<String> emptySet = java.util.Collections.emptySortedSet();
}
```

Apparently assigning the result to a variable or field causes the problem, as passing it as an argument does not fail.

**Expected behavior**
`java.util.Collections$EmptyNavigableSet` should not be reported as a missing class.

**Environment**
openjdk version "21.0.8" 2025-07-15 LTS
OpenJDK Runtime Environment Zulu21.44+17-CA (build 21.0.8+9-LTS)
OpenJDK 64-Bit Server VM Zulu21.44+17-CA (build 21.0.8+9-LTS, mixed mode, sharing)

SpotBugs Version: 4.10.1 and .2 and .3
